src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -0,0 +1,285 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Text;
+using System.Runtime.InteropServices;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Agent.Worker;
+using Microsoft.VisualStudio.Services.Agent.Worker.Build;
+using Microsoft.VisualStudio.Services.WebApi;
+using Build2 = Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.Win32;
+using System.Diagnostics;
+using System.Linq;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using Microsoft.VisualStudio.Services.Agent.Listener.Capabilities;
```

Comment:
you might want to move the stuff into the shared library, worker should not take dependency on agent. :)

Reply:
I can do that!

Reply:
@TingluoHuang Sorry I forgot to do this one. According to the Agent.Worker.csproj there already was a reference to the Agent:

```
<ProjectReference Include="..\Microsoft.VisualStudio.Services.Agent\Microsoft.VisualStudio.Services.Agent.csproj" />
```

The using statement you see is just the namespace for Agent.Listener. The CapabilitiesManager class lives in Agent.Listener and the namespace for that is Microsoft.VisualStudio.Services.Agent.Listener.Capabilities.

Let's chat this morning.

Reply:
Fixed.